internal/controller/finbackup_controller.go

```diff
@@ -666,13 +668,184 @@ func (r *FinBackupReconciler) createOrUpdateBackupJob(
 	return nil
 }
 
+//nolint:dupl
 func (r *FinBackupReconciler) createOrUpdateDeletionJob(ctx context.Context, backup *finv1.FinBackup) error {
-	// TODO: Implement the deletion job creation logic.
+	var job batchv1.Job
+	job.SetName(deletionJobName(backup))
+	job.SetNamespace(r.cephClusterNamespace)
+	if _, err := ctrl.CreateOrUpdate(ctx, r.Client, &job, func() error {
+		labels := job.GetLabels()
+		if labels == nil {
+			labels = map[string]string{}
+		}
+		labels["app.kubernetes.io/name"] = labelAppNameValue
+		labels["app.kubernetes.io/component"] = labelComponentDeletionJob
+		job.SetLabels(labels)
+
+		owners := []metav1.OwnerReference{
```

Comment:
Please remove ownerReference since cross-namepsace ownerReference is prohibited.

https://kubernetes.io/docs/concepts/overview/working-with-objects/owners-dependents/#owner-references-in-object-specifications